app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt

```diff
@@ -730,31 +758,52 @@ class NewMessageViewModel @Inject constructor(
         }.onFailure(Sentry::captureException)
     }
 
-    fun executeDraftActionWhenStopping(
-        action: DraftAction,
-        isFinishing: Boolean,
-        isTaskRoot: Boolean,
-        subjectValue: String,
-        uiBodyValue: String,
-        startWorkerCallback: () -> Unit,
-    ) = globalCoroutineScope.launch(ioDispatcher) {
+    fun saveBodyAndSubject(subject: String, html: String) {
+        globalCoroutineScope.launch(ioDispatcher) {
+            _subjectAndBodyChannel.send(Triple(subject, html, channelExpirationIdTarget))
+        }
+    }
 
-        val localUuid = draftLocalUuid ?: return@launch
+    fun discardOldBodyAndSubjectChannelMessages() {
+        channelExpirationIdTarget++
+    }
+
+    fun waitForBodyAndSubjectToExecuteDraftAction(draftSaveConfiguration: DraftSaveConfiguration) {
+        subjectAndBodyJob?.cancel()
+        subjectAndBodyJob = globalCoroutineScope.launch(ioDispatcher) {
+            val subject: String
+            val body: String
+            while (true) {
+                val (receivedSubject, receivedBody, expirationId) = subjectAndBodyChannel.receive()
+                if (expirationId == channelExpirationIdTarget) {
+                    subject = receivedSubject
+                    body = receivedBody
+                    break
+                }
+            }
+
+            draftSaveConfiguration.addSubjectAndBody(subject, body)
+            executeDraftAction(draftSaveConfiguration)
+        }
+    }
+
+    private suspend fun executeDraftAction(draftSaveConfiguration: DraftSaveConfiguration) = with(draftSaveConfiguration) {
```

Comment:
Make this function an extension of DraftSaveConfiguration

Reply:
After discussing about it, this object being only created right there for this only purpose makes it more suited for being passed as an argument